query-engine/query-engine/src/request_handlers/graphql/response.rs

```diff
@@ -111,3 +122,51 @@ impl From<CoreError> for GQLResponse {
         gql_response
     }
 }
+
+/// GQLBatchResponse converters
+
+impl GQLBatchResponse {
+    pub fn insert_responses(&mut self, responses: Vec<GQLResponse>) {
+        responses.into_iter().for_each(|response| {
+            self.batch_result.push(response);
+        })
+    }
+
+    pub fn insert_error(&mut self, error: impl Into<GQLError>) {
+        self.errors.push(error.into());
+    }
+}
+
+impl From<user_facing_errors::Error> for GQLBatchResponse {
+    fn from(err: user_facing_errors::Error) -> Self {
+        let mut batch_response = Self::default();
+        batch_response.insert_error(err);
+        batch_response
+    }
+}
+
+impl From<CoreError> for GQLBatchResponse {
+    fn from(err: CoreError) -> Self {
+        let mut batch_response = Self::default();
+
+        batch_response.insert_error(err);
+        batch_response
+    }
+}
+
+impl From<PrismaError> for GQLBatchResponse {
+    fn from(err: PrismaError) -> Self {
+        let mut responses = Self::default();
+        responses.insert_error(err);
+        responses
+    }
+}
+
+impl From<Vec<GQLResponse>> for GQLBatchResponse {
+    fn from(responses: Vec<GQLResponse>) -> Self {
+        let mut batch_response = Self::default();
+
+        batch_response.insert_responses(responses);
+        batch_response
+    }
+}
```

Comment:
I feel like this is not a great pattern, going from Vec<GQLResponse> to a GQLBatchResponse this way. Is there a better way to achieve that? I've been mostly trying to follow existing patterns in this part of the code.

Reply:
This is a common rust pattern, nothing wrong with it.